### PR TITLE
Issue 404: Add fix routine for L025

### LINKED ISSUE #404: Add fix routine for L025
[This query](http://sqlfluff-online.herokuapp.com/fluffed?sql=H4sIACarKV8C_ytOzUlNLlHQUkgrys9V0CiGcA0VEosVEjUVikuTCktTiyoBBIbljSYAAAA%3D&dialect=postgres) has two problems:

```sh
$ echo 'select * from (select 1 as a) subquery' | sqlfluff lint -
== [stdin] FAIL
L:   1 | P:  31 | L011 | Implicit aliasing of table not allowed. Use explicit
                       | `AS` clause.
L:   1 | P:  31 | L025 | Alias 'subquery' is never used in SELECT statement.
```

It gets fixed as such 

```sh
$ echo 'select * from (select 1 as a) subquery' | sqlfluff fix - --rules L011,L010,L025                                
select * from (select 1 as a) as subquery
```

But if you lint the fixed sql, L025 remains!

```sh
$ echo 'select * from (select 1 as a) subquery' | sqlfluff fix - --rules L011,L010,L025 | sqlfluff lint -
== [stdin] FAIL
L:   1 | P:  34 | L025 | Alias 'subquery' is never used in SELECT statement.
```

The conjunction of these two errors shows that the "real" fix would be [this query](http://sqlfluff-online.herokuapp.com/fluffed?sql=H4sIAEGrKV8C_ytOzUlNLlHQUkgrys9V0CiGcA0VEosVEjV5uQDXINLGHwAAAA%3D%3D&dialect=postgres)

```sh
$ echo 'select * from (select 1 as a)' | sqlfluff lint -
# no violations!
```

Should fluff remove the implicit alias if it is never used?

### PR DESCRIPTION
**Pull request by barrywhart**
Resolves #404

### CODE REVIEW

File: src/sqlfluff/core/dialects/dialect_ansi.py [hunk 1]
```diff
@@ -746,6 +747,16 @@ class FrameClauseSegment(BaseSegment):
 )
 
 
+class AliasInfo(NamedTuple):
```
Comment (barrywhart) [inline src/sqlfluff/core/dialects/dialect_ansi.py:750]: In working on this PR, I refactored some of these functions from using tuples to using NamedTuples. It makes the code easier to understand and maintain.

File: src/sqlfluff/core/rules/std/L015.py [hunk 2]
```diff
@@ -1,8 +1,10 @@
 """Implementation of Rule L015."""
 
 from ..base import BaseCrawler, LintFix, LintResult
+from ..doc_decorators import document_fix_compatible
 
 
+@document_fix_compatible
```
Comment (barrywhart) [inline src/sqlfluff/core/rules/std/L015.py:7]: This is not related to the main PR -- it corrects an oversight from my earlier PR (#732).

**Review by alanmcruickshank — APPROVED** [review]
LGTM 👍 

Nice work in using NamedTuples instead of the old implementation. Much cleaner 🎉

**Comment by nolanbconaway** [comment]
thannnks barry!

**Comment by barrywhart** [comment]
@nolanbconaway: No worries, it was fun! I didn't have time to work on the project much in 2020. Things have been calmer lately, and I'm really enjoying getting back into it.
